lib/active_merchant/billing/gateways/blue_snap.rb

```diff
@@ -412,9 +412,33 @@ def success_from(action, response)
         (200...300).cover?(response.code.to_i)
       end
 
-      def message_from(succeeded, parsed_response)
+      def message_from(succeeded, response)
         return 'Success' if succeeded
-        parsed_response['description']
+        parsed = parse(response)
```

Comment:
Since the definition on 530 belongs to a different class and neither inherits the other, they are effectively invisible to each other. It's a bit confusing to read/navigate, but since the implementation of `PaymentMethodDetails` is specific to this particular gateway, I think it makes sense for them to coexist in the same file.

Reply:
Oh, I hadn't noticed the different class contained in the same file. Thanks for pointing that out!